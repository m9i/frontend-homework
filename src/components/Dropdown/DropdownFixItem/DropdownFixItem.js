import React from "react";

const DropdownFixItem = ({text, icon}) => {
    const className = `nav__list__item ${icon === 'logout' ? 'color-red' : ''}`;

    return <div className={className}>
        <i className="material-icons-outlined nav__list__item--icon">
            {icon}
        </i>
        {text}
    </div>
}

export default DropdownFixItem;
