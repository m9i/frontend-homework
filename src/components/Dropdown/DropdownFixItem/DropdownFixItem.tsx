import React from 'react';
import {IDropdownFixItemProps} from '../../../interfaces/IDropdownFixItemProps';

const DropdownFixItem: React.FC<IDropdownFixItemProps> = ({text, icon}: IDropdownFixItemProps) => {
    const className: string = `nav__list__item ${icon === 'logout' ? 'color-red' : ''}`;

    return (
        <div className={className}>
            <i className="material-icons-outlined nav__list__item--icon">{icon}</i>
            {text}
        </div>
    );
};

export default DropdownFixItem;
