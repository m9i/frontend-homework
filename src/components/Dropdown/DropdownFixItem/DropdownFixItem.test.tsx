import React from 'react';
import DropdownFixItem from './DropdownFixItem';
import {shallow, mount, ReactWrapper, ShallowWrapper} from 'enzyme';
import {Provider} from 'react-redux';
import configureMockStore from 'redux-mock-store';

describe('DropdownFixItem', () => {
    const mockStore = configureMockStore([]);
    let store: any;

    beforeEach(() => {
        store = mockStore({
            company: {
                selectedCompany: {value: 'mock-name'},
            },
            dropdown: {
                showDropdown: false,
            },
        });
    });

    it('snapshot', () => {
        const DropdownFixItemComponent: ShallowWrapper = shallow(
            <Provider store={store}>
                <DropdownFixItem text="mock-text" icon="mock-icon"/>
            </Provider>
        ).dive();

        expect(DropdownFixItemComponent).toMatchSnapshot();
    });

    it('should set "color-red" css class for logout', () => {
        const DropdownFixItemComponent: ReactWrapper = mount(
            <Provider store={store}>
                <DropdownFixItem text="mock-text" icon="logout"/>
            </Provider>
        );
        const wrapper: ReactWrapper = DropdownFixItemComponent.find('div');

        expect(wrapper.hasClass('color-red')).toEqual(true);
    });

    it('should show item text', () => {

    });

    it('should show item icon', () => {

    });
});
