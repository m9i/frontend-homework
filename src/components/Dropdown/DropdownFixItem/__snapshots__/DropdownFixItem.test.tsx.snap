// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownFixItem snapshot 1`] = `
<DropdownFixItem
  icon="mock-icon"
  text="mock-text"
/>
`;
